runtime/interpreter/value.go

```diff
@@ -1487,7 +1487,13 @@ func (v *ArrayValue) IsImportable(inter *Interpreter) bool {
 }
 
 func (v *ArrayValue) checkInvalidatedResourceUse(interpreter *Interpreter, getLocationRange func() LocationRange) {
-	if v.isDestroyed || (v.array == nil && v.IsResourceKinded(interpreter)) {
+	if v.isDestroyed {
+		panic(DestroyedResourceError{
+			LocationRange: getLocationRange(),
+		})
+	}
```

Comment:
Wouldn't this hide the internal error?
 Maybe it's ok to treat that as an internal error since it is not supposed to reach here

Reply:
Treating use of a destroyed resource as an internal error will likely result in many reports and make the alert to noisy.

Reply:
It would not come to the runtime check under normal circumstances, wouldn't it? Use via references is already handled separately?

Reply:
Clarified offline